routes/books.js

```diff
@@ -0,0 +1,48 @@
+import express from "express";
+import booksData from "../data/books.json";
+
+const router = express.Router()
+
+router.get("/", (req, res) => {
+    res.json(booksData)
+})
+
+router.get("/top-rated", (req, res) => {
+    const highAvgRating = booksData.filter((book) => book.average_rating >= 4)
+    res.status(200).json(highAvgRating)
+})
+
+router.get("/title/:title", (req, res) => {
+    const title = req.params.title
+    let titleInput = booksData.find((str) => str.title.toLowerCase() === title.toLowerCase())
+
+    if (titleInput) {
+        res.status(200).json({
+            data: titleInput,
+            success: true
+        })
+    } else {
+        res.status(400).json({
+            data: "Not found",
+            success: false
+        })
+    }
+})
+
+router.get("/author/:authors", (req, res) => {
+    const authors = req.params.authors
+    let authorsInput = booksData.filter((str) => str.authors.toLowerCase() === authors.toLowerCase())
+    if (authorsInput) {
+        res.status(200).json({
+            data: authorsInput,
+            success: true
+        })
+    } else {
+        res.status(200).json({
+            data: "Not found",
+            success: false
+        })
+    }
```

Comment:
I tested out this endpoint in postman and I never seem to hit this else statement because the if statement on line 35 always seems to be true no matter what you type, even if you type an author that doesn't exist. 

Is there a reason why you would want to send the same 200 error code and the string "Not found" in the response body?

Reply:
Good question, I need to look into this. The reason why I wanted to send the 200 error code is because of the .filter. As I remember when you explained in the demo, it's a success request and the author name might be there in the future? I also wanted to return a collection of books with this author so filter seems to fit rather than find.

Thanks for pointing this out though. I will spend more time on this :)

Reply:
Fixed!! I just changed line 35 to (authorsInput.length > 0) :D 
The problem was that `authorsInput.` always return an array that's why it's always true. With that said, I checked the length of it instead hehe

Reply:
Yes! Sending a 200 like how Maksy suggested is what I would do as well in this situation. Sorry if I wasn't clear, I was just curious because usually a 200 status code would actually mean that the request was successful for example but that it would just return an empty array.

I think you could just omit the else statement if you still want to send a status code of 200 and successful true as the response would be successful but just return an empty array.

Reply:
Great job fixing line 35! Looks good! That makes sense..